crates/biome_analyze/CONTRIBUTING.md

```diff
@@ -354,37 +223,157 @@ pub enum Behavior {
 }
 ```
 
-### Deprecate a rule
+### Coding the rule
 
-There are occasions when a rule must be deprecated, to avoid breaking changes. The reason
-of deprecation can be multiple.
+Below, there are many tips and guidelines on how to create a lint rule using Biome infrastructure.
 
-In order to do, the macro allows adding additional field to add the reason for deprecation
+#### Navigating the CST
+
+Then navigating the nodes and tokens of certain nodes, will you notice straight away that the majority of those methods will return a `Result` (`SyntaxResult`).
+
+Generally, you will end up navigating the CST inside the `run` function, and this function will usually return an `Option` or a `Vec`.
+
+- If the `run` function returns an `Option`, you're encouraged to transform `Result` into `Option` and use the try operator `?`. This will make your coding way easier:
+  ```rust
+  fn run() -> Self::Signals {
+    let prev_val = js_object_member.value().ok()?;
+  }
+  ```
+- If the `run` function returns a `Vec`, you're encouraged to use the `let else` trick to reduce code branching:
+
+  ```rust
+    fn run() -> Self::Signals {
+      let Ok(prev_val) = js_object_member.value() else {
+        return vec![]
+      };
+    }
+  ```
+
+#### Query multiple nodes
+
+There are times when you might need to query multiple nodes at the time. Instead of querying the root of the CST, you can use the macro `declare_node_union!` to "join" multiple nodes into an `enum`:
```

Comment:
```suggestion
There are times when you might need to query multiple nodes at once. Instead of querying the root of the AST, you can use the macro `declare_node_union!` to "join" multiple nodes into an `enum`:
```